### PR TITLE
Add CISPO loss option and documentation

### PR DESCRIPTION
**Pull request by gustavorubim**
Introduce the CISPO loss type in GRPOConfig, enhancing the training process with truncated importance-sampling REINFORCE. Update documentation to explain the objective, usage, and related parameters, while adding logging metrics for better observability during training. This implementation supports the ScaleRL-inspired variant and improves the overall training experience.

### CODE REVIEW

File: trl/trainer/grpo_config.py [hunk 1]
```diff
@@ -538,22 +538,18 @@ class GRPOConfig(TrainingArguments):
...
+            "choices": ["grpo", "dapo", "bnpo", "dr_grpo", "cispo"],
         },
     )
+
+    cispo_clip_max: float = field(
```
Comment (pramodith) [inline trl/trainer/grpo_config.py:546]: I'd recommend re-using the `epsilon-high` config instead of creating a new one here.

File: trl/trainer/grpo_trainer.py [hunk 2]
```diff
@@ -1705,19 +1710,28 @@ def _compute_loss(self, model, inputs):
...
+        # level: "token" level -> (B, T); "sequence" level -> (B, 1)
         coef_1 = torch.exp(log_importance_weights)
-        coef_2 = torch.clamp(coef_1, 1 - self.epsilon_low, 1 + self.epsilon_high)
+        cispo_truncated_weights = None
+        cispo_clipped_mask = None
+
+        if self.loss_type == "cispo":
+            cispo_cap = torch.full_like(coef_1, self.cispo_clip_max)
+            cispo_truncated_weights = torch.minimum(coef_1, cispo_cap)
+            cispo_clipped_mask = coef_1 > cispo_cap
```
Comment (pramodith) [inline trl/trainer/grpo_trainer.py:1722]: Can be simplified by:

```
cispo_weights = torch.where(coef_1 < self.epsilon_high, coef_1, self.epsilon_high).detach()
```

**Comment by pramodith** [comment]
@gustavorubim just wanted to check if you've had the chance to make the requested changes.
